compiler+runtime/src/cpp/jank/analyze/expr/if.cpp

```diff
@@ -0,0 +1,44 @@
+#include <jank/analyze/expr/if.hpp>
+#include <jank/detail/to_runtime_data.hpp>
+
+namespace jank::analyze::expr
+{
+  using namespace jank::runtime;
+
+  if_::if_(expression_position const position,
+           local_frame_ptr const frame,
+           native_bool const needs_box,
+           expression_ptr const condition,
+           expression_ptr const then,
+           option<expression_ptr> const &else_)
+    : expression{ expr_kind, position, frame, needs_box }
+    , condition{ condition }
+    , then{ then }
+    , else_{ else_ }
+  {
+  }
+
+  void if_::propagate_position(expression_position const pos)
+  {
+    position = pos;
+    if(then)
+    {
+      then->propagate_position(pos);
+    }
+    if(else_.is_some())
+    {
+      else_.unwrap()->propagate_position(pos);
+    }
+  }
+
+  object_ptr if_::to_runtime_data() const
+  {
+    return merge(expression::to_runtime_data(),
+                 obj::persistent_array_map::create_unique(make_box("condition"),
+                                                          jank::detail::to_runtime_data(*condition),
+                                                          make_box("then"),
+                                                          jank::detail::to_runtime_data(*then),
+                                                          make_box("else"),
+                                                          jank::detail::to_runtime_data(else_)));
```

Comment:
Do we still want to assoc if `else_` is none? Ditto all the other `option<expression_ptr>` cases for other expressions.

Reply:
Yep, it'll render as `"none"`.

Reply:
Ah, does that mean none and an option containing the string `"none"` render the same?

Reply:
```cpp
  template <typename T>
  object_ptr to_runtime_data(option<T> const &m)
  {
    return obj::persistent_array_map::create_unique(
      make_box("__type"),
      make_box("option"),
      make_box("data"),
      (m.is_none() ? make_box("none") : to_runtime_data(m.unwrap())));
  }
```

Reply:
Looks like it? Might be an issue when converting data back to expressions.

Reply:
We don't convert data back to expressions. The whole `to_runtime_data` mechanism is for debugging AST expressions, used only by me. :joy: 

Once we can have a good Portal integration, hopefully turning jank's compiler data into Clojure data will prove much more useful.